LoRa Adapter crashes on startup: missing bean declaration

I guess the bean declaration for a _HttpAdapterMetrics_ bean is missing in its application configuration.

```
15:02:30.311 [main] ERROR o.e.h.a.l.i.Application$$EnhancerBySpringCGLIB$$f8610884 - exception occurred during startup, shutting down ...
java.util.concurrent.ExecutionException: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'loraProtocolAdapter' defined in class path resource [org/eclipse/hono/adapter/lora/impl/Config.class]: Unsatisfied dependency expressed through method 'loraProtocolAdapter' parameter 0; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.eclipse.hono.client.SendMessageSampler$Factory' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
	at org.eclipse.hono.service.AbstractBaseApplication.run(AbstractBaseApplication.java:161)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:775)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:765)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:322)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215)
	at org.eclipse.hono.adapter.lora.impl.Application.main(Application.java:37)
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'loraProtocolAdapter' defined in class path resource [org/eclipse/hono/adapter/lora/impl/Config.class]: Unsatisfied dependency expressed through method 'loraProtocolAdapter' parameter 0; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.eclipse.hono.client.SendMessageSampler$Factory' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:538)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1336)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1176)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:556)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:516)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:342)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.ObjectFactoryCreatingFactoryBean$TargetBeanObjectFactory.getObject(ObjectFactoryCreatingFactoryBean.java:153)
	at org.eclipse.hono.service.AbstractApplication.deployServiceVerticles(AbstractApplication.java:92)
	at org.eclipse.hono.service.AbstractApplication.lambda$deployVerticles$2(AbstractApplication.java:135)
	at io.vertx.core.Future.lambda$compose$3(Future.java:368)
	at io.vertx.core.impl.FutureImpl.dispatch(FutureImpl.java:105)
	at io.vertx.core.impl.FutureImpl.onComplete(FutureImpl.java:83)
	at io.vertx.core.Future.compose(Future.java:364)
	at io.vertx.core.Future.compose(Future.java:336)
	at org.eclipse.hono.service.AbstractApplication.deployVerticles(AbstractApplication.java:135)
	at org.eclipse.hono.service.AbstractBaseApplication.run(AbstractBaseApplication.java:154)
	... 6 common frames omitted
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.eclipse.hono.client.SendMessageSampler$Factory' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoMatchingBeanFound(DefaultListableBeanFactory.java:1717)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1273)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1227)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788)
	... 23 common frames omitted
15:02:30.372 [main] INFO  o.e.h.a.l.i.Application$$EnhancerBySpringCGLIB$$f8610884 - shutting down application...
```

--- a/adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/impl/Config.java
+++ b/adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/impl/Config.java
@@ -20,6 +20,7 @@ import javax.annotation.PostConstruct;
 
 import org.eclipse.hono.adapter.http.HttpAdapterMetrics;
 import org.eclipse.hono.adapter.http.HttpProtocolAdapterProperties;
+import org.eclipse.hono.adapter.http.MicrometerBasedHttpAdapterMetrics;
 import org.eclipse.hono.adapter.lora.LoraProtocolAdapterProperties;
 import org.eclipse.hono.adapter.lora.providers.LoraProvider;
 import org.eclipse.hono.client.SendMessageSampler;
@@ -38,6 +39,7 @@ import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Scope;
 
 import io.micrometer.core.instrument.MeterRegistry;
+import io.vertx.core.Vertx;
 
 /**
  * Spring Boot configuration for the LoRa adapter.
@@ -56,6 +58,11 @@ public class Config extends AbstractAdapterConfig {
             throw new IllegalStateException(
                     "LoRa Protocol Adapter does not support unauthenticated mode. Please change your configuration accordingly.");
         }
+    }
+
+    @Bean
+    HttpAdapterMetrics metrics(final MeterRegistry registry, final Vertx vertx) {
+        return new MicrometerBasedHttpAdapterMetrics(registry, vertx);
     }
 
     /**

--- a/site/homepage/content/release-notes.md
+++ b/site/homepage/content/release-notes.md
@@ -11,6 +11,7 @@ title = "Release Notes"
   now correctly indicate a server related problem instead.
 * The containers for the Device Registry implementations, the Authentication server, the Device Connection and
   Command Router services did not shut down gracefully upon receiving a SIGTERM signal. This has been fixed.
+* The LoRA protocol adapter failed to start due to a missing bean declaration. This has been fixed.
 
 ## 1.5.1
 

